mllib-local/src/main/scala/org/apache/spark/ml/linalg/Matrices.scala

```diff
@@ -999,7 +999,7 @@ object Matrices {
             val data = new ArrayBuffer[(Int, Int, Double)]()
             dnMat.foreachActive { (i, j, v) =>
               if (v != 0.0) {
-                data.append((i, j + startCol, v))
+                data.+=((i, j + startCol, v))
```

Comment:
You can write `data += ((i, j + startCol, v))`
Yes, I think it may be worth optimizing this because append actually expects varargs.
There are many more instances in the code though. I think it's reasonable to fix this everywhere. At least non-test code, and at least anything where performance could be important.